File: app/src/main/java/com/larryhsiao/nyx/jots/JotsDb.java
package com.larryhsiao.nyx.jots;

import com.silverhetch.clotho.Source;

import java.sql.Connection;

/**
 * Db connection source with initial sql.
 */
public class JotsDb implements Source<Connection> {
    private final Source<Connection> connSource;

    public JotsDb(Source<Connection> connSource) {
        this.connSource = connSource;
    }

    @Override
    public Connection value() {
        try {
            final Connection conn = connSource.value();
            conn.createStatement().executeUpdate(
                    // language=H2
                    "CREATE TABLE IF NOT EXISTS jots(" +
                            "id integer not null auto_increment," +
                            "content text not null, " +
                            "createdTime timestamp with time zone not null, " +
                            "location geometry " +
                            ");"
            );
            return conn;
        } catch (Exception e) {
            throw new IllegalArgumentException(e);
        }
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/jots/JotsDb.kt
package com.larryhsiao.nyx.core.jots

import com.silverhetch.clotho.Source
import java.sql.Connection

/**
 * Db connection source with initial sql.
 */
class JotsDb(private val connSource: Source<Connection>) : Source<Connection> {
    override fun value(): Connection {
        return try {
            val conn = connSource.value()
            conn.createStatement().executeUpdate(
                // language=H2
                """CREATE TABLE IF NOT EXISTS jots
(
    id          integer                  not null auto_increment,
    title       text                     not null default '',
    content     text                     not null,
    createdTime timestamp with time zone not null,
    location    geometry,
    mood        varchar                  not null default '',
    version     integer                  not null default 1,
    delete      integer                  not null default 0,
    private     boolean                  not null default false
);""")
            conn
        } catch (e: Exception) {
            throw IllegalArgumentException(e)
        }
    }
}